workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReadersWriters/ReaderWriterBase.cs

```diff
@@ -9,16 +9,15 @@
 namespace Improbable.Gdk.Core.GameObjectRepresentation
 {
     internal abstract class ReaderWriterBase<TSpatialComponentData, TComponentUpdate>
-        : IWriter<TSpatialComponentData, TComponentUpdate>
+        : RequirableBase
```

Comment:
Won't removing the `IWriter` inheritance cause problems?

Reply:
Stuff seems to be still running fine because the API code doesn't depend on `ReaderWriterBase` inheriting IWriter. The final generated Impl class inherits from everything since it inherits the codegened Reader/Writer interfaces which themselves ihnehrit from the general IReader/IWriter interface. I suppose this comes down to a matter of style. I could let ReaderWriterBase inherit from IReader and IWriter if we want but it's not necessary.

Reply:
No reason why it couldn't implement the IWriter interface while also having a parent class though surely? Which would make more sense if this is where the actual method is being implemented

Reply:
Let me just let `ReaderWriterBase` from IReader and IWriter then. Why the hell not? ;)